qa/sc_evm_forging_fee_payments.py

```diff
@@ -0,0 +1,323 @@
+      assert_equal(
+          sc_node_2.wallet_getTotalBalance()['result']['balance'],
+          ft_amount_in_wei -
+          (forger_stake_amount_in_wei + transactionFee_0) -
+          (transferred_amount_in_wei_1 + transactionFee_1) -
+          (transferred_amount_in_wei_2 + transactionFee_2)
+      )
+
+      # Collect SC node balances before fees redistribution
+      sc_node_1_balance_before_payments = sc_node_1.wallet_getTotalBalance()['result']['balance']
+      sc_node_2_balance_before_payments = sc_node_2.wallet_getTotalBalance()['result']['balance']
+
+      # Generate one more block with no fee by SC node 2 to reach the end of the withdrawal epoch
+      sc_last_we_block_id = generate_next_block(sc_node_2, "second node")
+      sc_block_fee_info.append(BlockFeeInfo(2, 0, 0))
+
+      self.sc_sync_all()
+
+      # Collect fee values
+      total_fee = 0
+      pool_fee = 0.0
+      forger_fees = {}
+      for sc_block_fee in sc_block_fee_info:
+          total_fee += sc_block_fee.baseFee + sc_block_fee.forgerTips
+          pool_fee += sc_block_fee.baseFee
+
+      print("total fee = {}".format(total_fee))
+      print("pool fee = {}".format(pool_fee))
+
+      for idx, sc_block_fee in enumerate(sc_block_fee_info):
+          if sc_block_fee.node in forger_fees:
+              forger_fees[sc_block_fee.node] += sc_block_fee.forgerTips
+          else:
+              forger_fees[sc_block_fee.node] = sc_block_fee.forgerTips
+
+          forger_fees[sc_block_fee.node] += math.floor(pool_fee / len(sc_block_fee_info))
+
+          if idx < pool_fee % len(sc_block_fee_info):
+              forger_fees[sc_block_fee.node] += 1
+
+          print("block {} fees: {}, {}".format(idx, sc_block_fee.baseFee, sc_block_fee.forgerTips))
+
+
+      sc_node_1_balance_after_payments = sc_node_1.wallet_getTotalBalance()['result']['balance']
+      sc_node_2_balance_after_payments = sc_node_2.wallet_getTotalBalance()['result']['balance']
+
+      node_1_fees = forger_fees[1]
+      node_2_fees = forger_fees[2]
+
+      # Check forger fee payments
+      print("SC1 bal before = {}, after = {}".format(sc_node_1_balance_before_payments, sc_node_1_balance_after_payments))
+      print("SC2 bal before = {}, after = {}".format(sc_node_2_balance_before_payments, sc_node_2_balance_after_payments))
+
+      print("SC1 forger fees = {}".format(node_1_fees))
+      print("SC2 forger fees = {}".format(node_2_fees))
+
+      assert_equal(node_1_fees + node_2_fees, total_fee)
+
+      # Check forger fee payments
+      assert_equal(sc_node_1_balance_after_payments, sc_node_1_balance_before_payments + node_1_fees,
+                   "Wrong fee payment amount for SC node 1")
+      assert_equal(sc_node_2_balance_after_payments, sc_node_2_balance_before_payments + node_2_fees,
+                   "Wrong fee payment amount for SC node 2")
+
+      '''
+      TODO enable when API will be ready
```

Comment:
do you plan to implement API in this PR or as a part of the separate task after?

Reply:
Yes correct, I added it in PR #531